source/lib/JMSLab/builders/build_latex.py

```diff
@@ -44,3 +46,72 @@ def add_call_args(self):
                                   os.path.normpath(self.log_file))
         self.call_args = args
         return None
+     
+    def add_bib_name(self, target):
+        if bool(target):
+            target_file = os.path.splitext(str(target[0]))[0]
+        else:
+            target_file = ''
+        self.bib_name = target_file
+        return None
+        
+    def check_bib(self, source):
+    
+        bibext = '.bib'
+        bib_file = ''
+        
+        if bool(source):
+            sources = misc.make_list_if_string(source)
+            for i in range(len(source)):
+                source_file = str(source[i])
+                if bibext in source_file:
+                    bib_file = source_file
+        else:
+            bib_file = ''  
+        self.check_bib = bib_file
+        return None
```

Comment:
Minor stuff (not essential):

- To avoid confusion in the future, it might be a good idea to make `self.check_bib` a boolean rather than a file name (it should be possible to have multiple `.bib` files; if `check_bib` is a boolean then it's clear it's not the name of the `.bib` files but a T/F indicator).
- Just as an fyi you can do `for source_file in source:` here.